Source/WebCore/rendering/svg/SVGRenderSupport.h

```diff
@@ -46,6 +47,8 @@ class SVGRenderSupport {
 public:
     static void layoutDifferentRootIfNeeded(const RenderElement&);
 
+    static bool isEmptySVGInlineText(const RenderObject*);
```

Comment:
I know you've cherry-picked a Blink change, but I suggest to not add anything new, valuable to SVGRenderSupport, which is a messy legacy container class. This should go in another place.
I'd vote for using dynamicDowncast<> on the call side, and just avoid the helper alltogether.

Reply:
I looked into Blink code and they have something like:

Or I am not clear on 'another place', any idea, where I could move this?

Blink latest code look something like this:

```
bool SVGRenderSupport::isRenderableTextNode(const RenderObject* object)
{
    ASSERT(object->isText());
    // <br> is marked as text, but is not handled by the SVG layout code-path.
    const auto* svgInlineText = dynamicDowncast(object);
    return svgInlineText && !svgInlineText->hasEmptyText();
}
```

Reply:
Only `RenderSVGInline` and `RenderSVGText` use the new helper method. Both renderers share the same base class, `RenderLayerModelObject`, where we recently added SVG specific code with the intent to share it between SVG renderers that do not inherit from RenderSVGModelObject (which is the next best place to share code between related SVG renderers).

RenderSVGInline inherits from RenderInline : RenderBoxModelObject : RenderLayerModelObject : RenderElement : RenderObject, and RenderSVGText inherits from RenderSVGBlock : RenderBlockFlow : RenderBlock : RenderBox : RenderBoxModelObject : RenderLayerModelObject... Therefore RenderLayerModelObject is the nearest common ancestor.

I'd suggest to move the code there (there's already LBSE specific code there).

Reply:
Just a note to myself for fixing that I can use 'lambda' like:
```
auto isEmptySVGInlineText = [](const RenderObject* object) {
        const auto svgInlineText =  dynamicDowncast<RenderSVGInlineText>(object);
        return svgInlineText && svgInlineText->hasEmptyText();
    };
```